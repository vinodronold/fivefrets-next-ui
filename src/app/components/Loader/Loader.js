import React, { PureComponent } from 'react'
import { color } from '../../constants/theme'
import { Caption } from '../Typography'

export default class Loader extends PureComponent {
  render() {
    return (
      <div className="container">
        <div className="loader">
          <div className="bar" />
          <div className="bar d1" />
          <div className="bar d2" />
          <div className="bar d3" />
          <div className="bar d4" />
        </div>
        <Caption>LOADING...</Caption>

        <style jsx>{`
          .container {
            text-align: center;
          }
          .loader {
            display: flex;
            justify-content: center;
            margin: 2rem;
          }
          .bar {
            padding: 1rem;
            margin: 0.25rem;
            background-color: ${color.secondary()};
            transform: scaleY(0.75);
            animation: loading 0.5s infinite ease-in-out alternate;
          }
          .d1 {
            animation-delay: 0.1s;
          }
          .d2 {
            animation-delay: 0.2s;
          }
          .d3 {
            animation-delay: 0.3s;
          }
          .d4 {
            animation-delay: 0.4s;
          }
          @keyframes loading {
            0% {
              transform: scaleY(0.75);
            }
            100% {
              transform: scaleY(1.25);
            }
          }
        `}</style>
      </div>
    )
  }
}
