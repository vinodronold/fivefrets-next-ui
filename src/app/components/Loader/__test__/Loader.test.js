import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Loader from '../Loader'

describe('components/Loader.js', () => {
  let _loader = shallow(<Loader />)
  it('should contain one container class', () => {
    expect(_loader.find('.container')).toHaveLength(1)
  })
  it('should contain one loader class', () => {
    expect(_loader.find('.loader')).toHaveLength(1)
  })
  // it('should contain five bar class', () => {
  //   expect(_loader.find('.bar')).toHaveLength(5)
  // })
  // it('should contain one d1 d2 d3 d4 class', () => {
  //   expect(_loader.find('.d1')).toHaveLength(1)
  //   expect(_loader.find('.d2')).toHaveLength(1)
  //   expect(_loader.find('.d3')).toHaveLength(1)
  //   expect(_loader.find('.d4')).toHaveLength(1)
  // })
  it('should matah the snapshot', () => {
    let _loader = renderer.create(<Loader />)
    expect(_loader).toMatchSnapshot()
  })
})
