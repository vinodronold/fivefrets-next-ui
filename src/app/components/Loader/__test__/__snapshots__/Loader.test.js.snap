// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Loader.js should match the snapshot 1`] = `
<div
  className="jsx-3649670468 container"
>
  <div
    className="jsx-3649670468 loader"
  >
    <div
      className="jsx-791847780 jsx-2291006144 bar"
    />
    <div
      className="jsx-791847780 jsx-3500035231 bar"
    />
    <div
      className="jsx-791847780 jsx-1901401244 bar"
    />
    <div
      className="jsx-791847780 jsx-40336921 bar"
    />
    <div
      className="jsx-791847780 jsx-3370661658 bar"
    />
  </div>
  <div
    className="jsx-4247411757 jsx-3754337244"
    style={Object {}}
  >
    LOADING...
  </div>
</div>
`;
